source/slang/slang-emit-c-like.cpp

```diff
@@ -3906,13 +3906,20 @@ void CLikeSourceEmitter::emitVecNOrScalar(
         // In other languages, we can output the Slang vector type directly
         else
         {
-            emitType(vectorType);
+            if (getTarget() == CodeGenTarget::WGSL)
```

Comment:
why is this needed? Shouldn't `emitType` be a virtual call into `WGSLSourceEmitter::emitTypeImpl` or something that handles the emit of vectorType properly?